packages/js-sdk/src/ConstantFlowAgreementV1Helper.d.ts

```diff
@@ -3,16 +3,64 @@ import type { Framework } from "./Framework";
 import type { LoadedContract } from "./loadContracts";
 import type BN from 'bn.js';
 
-export type Flow = {
+// returned from listFlows
+export interface FlowList {
+  inFlows: {
+    sender: string;
+    receiver: string;
+    flowRate: string;
+  }[];
+  outFlows: {
+    sender: string;
+    receiver: string;
+    flowRate: string;
+  }[];
+}
+
+// from CFAv1 contract
+// returned from _sanitizeflowInfo
+// returned from getFlow and getAccountFlowInfo
+export interface FlowInfo {
+  timestamp: Date;
+  flowRate: string;
+  deposit: string;
+  owedDeposit: string;
+}
+
+// Same interface used for create and update flow
+export interface CreateFlowOptions {
+  superToken: string;
   sender: string;
   receiver: string;
   flowRate: string;
-};
+  userData?: string;
+  onTransaction?: ()=>any
+}
+export type UpdateFlowOptions = CreateFlowOptions
 
-export type ListFlowsType = Promise<{
-  inFlows: Array<Flow>;
-  outFlows: Array<Flow>;
-}>;
+// MUST include 'by' type, it's only diff between create and update
+export interface DeleteFlowOptions {
+  superToken: string;
+  sender: string;
+  receiver: string;
+  flowRate: string;
+  by: string;
```

Comment:
I don't think `by` should be required, if you look at L154-155 of `CFAv1Helper.js`. It handles the case where it can be undefined (just uses the sender address for `by`).